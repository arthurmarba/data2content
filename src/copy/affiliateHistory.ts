export const STATUS_LABEL: Record<string, string> = {
  pending: 'Pendente',
  available: 'Disponível',
  paid: 'Pago',
  canceled: 'Cancelado',
  reversed: 'Revertido',
};

export const REASON_LABEL: Record<string, string> = {
  refund_within_hold: 'Reembolso dentro do período de 7 dias.',
  refund_after_payout: 'Reembolso após liberação (ajuste aplicado).',
  chargeback: 'Estorno por contestação do pagamento.',
  adjustment_after_redeem: 'Ajuste pós-resgate.',
  self_referral_blocked: 'Autoindicação não é elegível a comissão.',
  coupon_100: 'Pagamento com 100% de desconto não gera comissão.',
  trial_no_commission: 'Período de teste gratuito não gera comissão.',
  currency_mismatch: 'Moeda incompatível para saque.',
  duplicate_prevented: 'Entrada duplicada ignorada.',
  admin_manual_correction: 'Correção manual pela equipe.',
};

export function humanizeReason(code?: string | null): string {
  if (!code) return 'Ajuste no registro.';
  return REASON_LABEL[code] ?? 'Ajuste no registro.';
}
