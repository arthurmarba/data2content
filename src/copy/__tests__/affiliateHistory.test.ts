import { STATUS_LABEL, REASON_LABEL, humanizeReason } from '../affiliateHistory';

describe('affiliateHistory copy maps', () => {
  test('status labels snapshot', () => {
    expect(STATUS_LABEL).toMatchSnapshot();
  });

  test('reason labels snapshot', () => {
    expect(REASON_LABEL).toMatchSnapshot();
  });

    test('humanizeReason fallback', () => {
      expect(humanizeReason('unknown')).toBe('Ajuste administrativo.');
      expect(humanizeReason()).toBe('Ajuste administrativo.');
    });
});
