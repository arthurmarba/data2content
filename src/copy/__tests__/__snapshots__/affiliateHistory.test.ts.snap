// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`affiliateHistory copy maps reason labels snapshot 1`] = `
{
  "adjustment_after_redeem": "Ajuste pós-resgate.",
  "admin_manual_correction": "Correção manual pela equipe.",
  "chargeback": "Estorno por contestação do pagamento.",
  "coupon_100": "Pagamento com 100% de desconto não gera comissão.",
  "currency_mismatch": "Moeda incompatível para saque.",
  "duplicate_prevented": "Entrada duplicada ignorada.",
  "refund_after_payout": "Reembolso após liberação (ajuste aplicado).",
  "refund_within_hold": "Reembolso dentro do período de 7 dias.",
  "self_referral_blocked": "Autoindicação não é elegível a comissão.",
  "trial_no_commission": "Período de teste gratuito não gera comissão.",
}
`;

exports[`affiliateHistory copy maps status labels snapshot 1`] = `
{
  "available": "Disponível",
  "canceled": "Cancelado",
  "paid": "Pago",
  "pending": "Pendente",
  "reversed": "Revertido",
}
`;
